support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+                if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                    echo "Setting bucket to ${bucket_name}"
+                    bucket_name="${bucket_name//\"}"
+                    break
+                fi
+            fi
+        fi
+    done
+
+    if [[ -z ${bucket_name:-} ]]; then
+      echo ""
+      echo "Please enter a target bucket name and press [ENTER]:"
+      read -r bucket_name
+    fi
+
+    # Check if bucket exists
+    if [ "${s3type}" == 'minio' ]; then
+        if checkBucket "${AWS_REGION}" "${bucket_name}" >/dev/null; then
+            echo ""
+            echo "Bucket: ${bucket_name} found!"
+            echo ""
+            read -r -p "Are you sure you would like to use this bucket? [y/N] " response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                echo "Using specified bucket."
+                export S3_BUCKET=${bucket_name}
+            else
+                setBucket
+            fi
+        else
+            echo "Bucket: ${bucket_name} not found!"
+            echo "Please specify a different bucket and try again."
+            setBucket
+        fi
+    else
+        if aws s3api list-objects --bucket "${bucket_name}" --region "${AWS_REGION}" >/dev/null; then
+            echo "Bucket: ${bucket_name} found!"
+            echo "WARNING: Specified bucket is not empty!"
+            read -r -p "Are you sure you would like to use this bucket? [y/N] " response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                echo "Using specified bucket."
+                export S3_BUCKET=${bucket_name}
+            else
+                setBucket
+            fi
+        else
+            echo "Bucket: ${bucket_name} not found!"
+            echo "Please specify a different bucket and try again."
+            setBucket
+        fi
+    fi
+}
+
+checkBucket() {
+    region="${1}"
+    bucket="${2}"
+    aws --endpoint-url "${region}" s3api list-objects --bucket "${bucket}"
+}
+
+setRegion() {
+    # Sets the region used by the bucket
+    if [ "${s3type}" == 'minio' ]; then
+        sgroups=("default" "dev" "prod" "acceptance" "live" "blue" "green")
+        for i in "${sgroups[@]}"; do
+            if curl -s localhost:9631/services/builder-minio/"${i}" > /dev/null; then
+                minioIP=$(curl -s localhost:9631/services/builder-minio/"${i}" | jq .sys.ip)
+                echo ""
+                echo "We've detected your minio instance at: ${minioIP}!"
+                echo ""
+                read -r -p "Would you like to use this minio instance? [y/N] " response
+                if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                    echo "Setting endpoint to ${minioIP//\"}:9000"
+                    if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.use_ssl | grep "true"> /dev/null; then
+                        export AWS_REGION="https://${minioIP//\"}:9000"
```

Comment:
nope, not in this case. Minio has no concept of URIs, we use the "region" interchangeably with "endpoint-url" simply because its the closest similar abstraction in minio that we can key off of.